api/src/services/lottery.service.ts

```diff
@@ -224,4 +289,702 @@ export class LotteryService {
+                  return preference;
+                } else if (parseProgram) {
+                  // curr should equal the preference id we're pulling from
+                  if (!programs) {
+                    programs =
+                      (app.programs as unknown as ApplicationMultiselectQuestion[]) ||
+                      [];
+                  }
+                  parsePreference = false;
+                  // there aren't typically many programs, but if there, then a object map should be created and used
+                  const program = programs.find(
+                    (preference) => preference.key === curr,
+                  );
+                  multiselectQuestionValue = true;
+                  return program;
+                }
+
+                // sets parsePreference to true, for the next iteration
+                if (curr === 'preferences') {
+                  parsePreference = true;
+                } else if (curr === 'programs') {
+                  parseProgram = true;
+                }
+
+                if (acc === null || acc === undefined) {
+                  return '';
+                }
+
+                // handles working with arrays, e.g. householdMember.0.firstName
+                if (!isNaN(Number(curr))) {
+                  const index = Number(curr);
+                  return acc[index];
+                }
+
+                return acc[curr];
+              }, app);
+              value = value === undefined ? '' : value === null ? '' : value;
+              if (header.format) {
+                value = header.format(value);
+              }
+
+              row[`${header.path}`] = value ? value.toString() : '';
+            });
+            rows.push(row);
+          });
+          resolve(rows);
+        }),
+      );
+    }
+    const res = await Promise.all(promiseArray);
+
+    // add rows to spreadsheet
+    res.forEach((elem) => {
+      spreadsheet.addRows(elem);
+    });
+  }
+
+  buildExportColumns(
+    csvHeaders: CsvHeader[],
+    preference?: IdDTO,
+  ): Partial<Column>[] {
+    const res: Partial<Column>[] = csvHeaders.map((header) => ({
+      key: header.path,
+      header: header.label,
+    }));
+
+    if (preference) {
+      const indx = res.findIndex(
+        (header) => header.header === 'Raw Lottery Rank',
+      );
+
+      res[indx].header = `${preference.name} Rank`;
+
+      res.splice(indx, 0, {
+        key: 'Raw Lottery Rank',
+        header: 'Raw Lottery Rank',
+      });
+    }
+    return res;
+  }
+
+  getActivityLogKey = (logData: Prisma.JsonValue | null) => {
+    if (!logData) return null;
+    return logData[Object.keys(logData)[0]];
+  };
+
+  getLotteryStatusFromActivityLogMetadata = (
+    status: LotteryActivityLogStatus,
+    index: number,
+    previousStatus: LotteryActivityLogStatus | null,
+  ): LotteryActivityLogStatus => {
+    if (!status) return;
+    if (index === 0) return status;
+
+    if (
```

Comment:
This is not specific to this release branch, but I just noticed it so calling it out. When a lottery is rerun after it has been released it also goes back to the "ran" state. And the history log here displays it as "retracted". It is "retracted" but it also reran so I'm curious what we should be displaying here

Reply:
Good q! Will ask product